pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -811,12 +803,6 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 			}
 			for _, service := range services {
 				for _, servicePort := range service.Ports {
-					// check if this node is capable of starting a listener on this service port
-					// if bindToPort is true. Else Envoy will crash
-					if !validatePort(node, servicePort.Port, bindToPort) {
```

Comment:
What are the implications of removing this?

Reply:
implications = easier to use, less head scratching on why my sidecar fails to listen on port 80 even after me explicitly telling Istio to do so. This check should not have been there in the first place. If I am running the sidecar on a VM with fork+exec+other perms that allow me to launch sidecar on port 80, I should not be required to figure out some god awful intricate detail in pilot that checks for some UID==0 [which I am not running as], to decide that herr pilot needs me to fake my UID as 0 to get this POS workign

Reply:
Well, more head scratching for users with mesh expansion who will have their app crashing or breaking at some random point, when a config is pushed to them. 

1. Can you confirm that sending a port 80 listener to an envoy running as non-root will not prevent startup of other ports ? Including that the entire LDS is not rejected if port 80 can't be used ? 

2. If you have a special kernel or some magic that allows you to use port 80, you can indicate it to pilot - no need to fake your UID, you can add an explicit config to indicate the capability of the system and how it should get bind to port 80. There are several ways, including dropping root.

3. I understand you want your use case to work - but can you consider the impact on production users who may be affected - and test that they'll not be broken by this change ? I'm pretty sure the behavior of envoy when it can't bind to a port was affecting the other ports and resulted in the LDS rejection. 

Since this is risky - don't we have a rule that we should have flags to allow people to turn it on/off, and some test that
a machine without root getting port 80 will still work on the other ports ?